src/App.tsx

```diff
@@ -1,24 +1,227 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import {
+  getTodos,
+  addTodoOnServer,
+  deleteTodoFromServer,
+  updateTodoOnServer,
+} from './api/todos';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
+import { Status } from './enums/Status';
 
-const USER_ID = 0;
+const USER_ID = 7010;
+
+function filteredTodos(todos: Todo[], status: Status) {
+  if (status !== Status.All) {
+    return [...todos].filter(todo => {
+      return status === Status.Completed
+        ? todo.completed
+        : !todo.completed;
+    });
+  }
+
+  return todos;
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState(Status.All);
+  const [hasError, setError] = useState('');
+  const [isLoading, setLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingIds, setLoadingIds] = useState<Set<number>>(new Set());
+
+  const addLoadingTodo = useCallback((id: number) => {
+    setLoadingIds(state => {
+      state.add(id);
+
+      return new Set(state);
+    });
+  }, []);
+
+  const removeLoadingTodo = useCallback((id: number) => {
+    setLoadingIds(state => {
+      state.delete(id);
+
+      return new Set(state);
+    });
+  }, []);
+
+  const isTodoLoading = useCallback(
+    (id: number) => loadingIds.has(id),
+    [loadingIds],
+  );
+
+  const loadTodos = useCallback(
+    async () => {
+      try {
+        const loadedTodos = await getTodos(USER_ID);
+
+        setTodos(loadedTodos);
+        setError('');
+      } catch {
+        setError('Unable to add a todo');
+        setTimeout(() => setError(''), 3000);
+      }
+    }, [getTodos],
+  );
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const vidibleTodos = useMemo(() => {
+    return filteredTodos(todos, status);
+  }, [todos, status]);
+
+  const addTodo = useCallback(
+    async (title: string) => {
+      setLoading(true);
+      if (!title) {
+        setError('Title can t be empty');
+        setTimeout(() => setError(''), 3000);
+      }
+
+      const newTodo = {
+        id: 0,
+        userId: USER_ID,
+        title,
+        completed: false,
+      };
+
+      setTempTodo(newTodo);
+
+      try {
+        await addTodoOnServer(newTodo);
+        setTempTodo(newTodo);
```

Comment:
you already did this few lines before

Reply:
This is not done yet, `setTempTodo` is called a few lines higher